apps/aechannel/src/aesc_listeners.erl

```diff
@@ -177,63 +191,41 @@ maybe_close_lsock(Port, LSock, #st{ ports = Ports, socks = Socks } = St) ->
                  , socks = Socks1 }
     end.
 
-
 new_tab(Name) ->
     db_new(Name, [{keypos, 2}, ordered_set, public, named_table]).
 
 db_new(Name, Opts) ->
     {_, P} = lists:keyfind(keypos, 1, Opts),
-    #{name => Name, keypos => P, db => []}.
+    #{name => Name, keypos => P, db => ets:new(Name, Opts)}.
 
-db_lookup(#{db := L, keypos := P}, Key) ->
-    case lists:keyfind(Key, P, L) of
-        false ->
-            [];
-        Other ->
-            [Other]
-    end.
+db_lookup(#{db := Tab, keypos := _P}, Key) ->
+    ets:lookup(Tab, Key).
 
-db_delete(#{db := L, keypos := P} = Db, Key) ->
-    Db#{db => lists:keydelete(Key, P, L)}.
+db_delete(#{db := Tab, keypos := P} = Db, Key) ->
+    ets:delete(Tab, Key),
+    Db.
 
-db_insert(#{keypos := P, db := L} = Db, Objs) when is_list(Objs) ->
-    Db#{db => lists:foldl(fun(O, Lx) ->
-                                  lists:keystore(element(P, O), P, Lx, O)
-                          end, L, Objs)};
-db_insert(#{db := L, keypos := P} = Db, Obj) ->
-    K = element(P, Obj),
-    Db#{db => lists:keystore(K, P, L, Obj)}.
+db_insert(#{keypos := _P, db := Tab} = Db, Objs) when is_list(Objs) ->
+    ets:insert(Tab, Objs),
+    Db;
+db_insert(#{db := Tab, keypos := _P} = Db, Obj) ->
+    ets:insert(Tab, Obj),
+    Db.
 
-db_next(#{db := L, keypos := P}, Key) ->
-    db_next_(L, P, Key).
+db_next(#{db := Tab, keypos := P}, Key) ->
+    ets:next(Tab, Key).
 
-db_next_([H|T], P, K) when element(P, H) =:= K ->
-    case T of
-        [] -> '$end_of_table';
-        [Next|_] -> element(P, Next)
-    end;
-db_next_([_|T], P, K) ->
-    db_next_(T, P, K);
-db_next_([], _, _) ->
-    '$end_of_table'.
+db_select(#{db := Tab}, Pat) ->
+    ets:select(Tab, Pat).
 
-list_pids(Ports, P) ->
-    list_pids_(db_next(Ports, {P, 0}), P, Ports).
-
-list_pids_({P, Pid} = K, P, Ports) ->
-    [Pid | list_pids_(db_next(Ports, K), P, Ports)];
-list_pids_(_, _, _) ->
-    [].
+list_pids(#{db := PortsTab}, P) ->
+    ets:select(PortsTab, [{ #port{ key = {P,'$1'}, _ = '_'}, [], ['$1'] }]).
 
 get_lsock_info(pids, #{port := Port} = I, #st{ ports = Ports }) ->
     I#{pids => list_pids(Ports, Port)};
-get_lsock_info(responder, #{port := Port} = I, #st{ ports = Ports }) ->
-    case db_lookup(Ports, {Port, 0}) of
-        [#port{responder = R}] ->
-            I#{responder => R};
-        [] ->
-            I
-    end;
+get_lsock_info(responders, #{port := Port} = I, #st{ responders = Resps }) ->
```

Comment:
Could you please review `responder` in the `all` clause below?

Reply:
Shall that `all` piece of functionality be deleted?

Reply:
Let's keen the `all` option available for now, e.g. for debugging purposes. But I've pushed a fix for the `responder -> responders` bug.